docs/DeveloperGuide.adoc

```diff
@@ -142,105 +142,706 @@ image::StorageClassDiagram.png[]
 The `Storage` component,
 
 * can save `UserPref` objects in json format and read it back.
-* can save the Address Book data in json format and read it back.
+* can save the Address Book, History Manager, Finance Tracker, Cca Tracker, Course Planner and Planner
+data in json format and read it back.
 
 [[Design-Commons]]
 === Common classes
 
-Classes used by multiple components are in the `seedu.addressbook.commons` package.
+Classes used by multiple components are in the `seedu.jarvis.commons` package.
 
-== Implementation
+== Implementation==== Implementation
 
 This section describes some noteworthy details on how certain features are implemented.
 
 // tag::undoredo[]
-=== [Proposed] Undo/Redo feature
-==== Proposed Implementation
+=== Undo/Redo Feature
+==== Overview
+The application should be able to undo and redo changes made by commands to give the user more flexibility in their inputs.
+Undo and redo operations should also be undo or redo multiple commands in a command.
+In the event that a undo/redo command that comprises of multiple undo/redo operations fails at any point, all changes made
+by the command should be rolled back. This is reflected in the Activity Diagram below:
+
+.Activity Diagram for undo and redo commands
+image::UndoRedoActivityDiagram.png[]
```

Comment:
The actor can be omitted from the descriptions (instead of 'User enters', just start with 'Enters') to shorten the sentences. Also, both diagrams are very similar, maybe it would be better to remove the redo diagram and either change the undo diagram to include redo ('enters an undo/redo command', etc.) or talk about it in the paragraph right afterwards.